### DISCUSSION TITLE
🌊 Crabada.py now compatible with the Swimmer network!

### DISCUSSION THREAD

**Discussion by coccoinomane** [coccoinomane/crabada.py#104] [Announcements]
Hello snib-snibbers 🦀

Thanks to the valiant efforts of @EugeneTeu and @yigitest, the bot is now compatible with Swimmer Network!

Running the bot on Swimmer requires minimal changes, as detailed in the [🌊 Swimmer Network Tutorial](https://github.com/coccoinomane/crabada.py/wiki/%F0%9F%8C%8A-Swimmer-Network-tutorial) in the Wiki.

Make sure to follow the tutorial, and comment below if you encounter any issues 🙂

Cheers,
Cocco

**Comment by nikohd12** [comment]
Thank you so much @coccoinomane and the rest of the open-source squad @yigitest @loayei @EugeneTeu and others. 

Can confirm that the new bot works on swimmer after 24+ hrs of testing.

Whoever reads this, consider donating or contributing to this wonderful project.

**Comment by MaroYqr** [comment]
First thanks you all for the making changes so that the bots works on swimmer . I have tried it , everything is working normally for me except reinforcement part . I choose the highestMp strategy , it keep returning crab with highest BP , tried it with highestcosthighmp strategy , also returning with highest bp 237 crab . Anyone expriencing this ?

**Reply by yigitest** [reply]
Due to the limits from the crabada-api and the growing market, we cannot list all the options as before. 

HighestMp gives you the highest MP from the top 100 of the cheapest crabs. It may even be that no prime or craboid came from the query. 

If you are comfortable with editing python files yourself see this answer: https://github.com/coccoinomane/crabada.py/discussions/96#discussioncomment-2679497 You can limit the query with primes or craboids if high mp is what you want. Other than that I'll deal with this the first time I'm free but you'll have to wait.

**Reply by MaroYqr** [reply]
The code was fine , everything working fine .
Found the my mistake in .env 

Been using the wrong parameter name
USER1_GROUP1_TEAMS_REINFORCEMENT_STRATEGY

Corrected to 
USER1_GROUP1_REINFORCEMENT_STRATEGY

Once again thank you for the update 😬🙏

**Comment by Simosa1080** [comment]
Is anyone else have the problem this problem that it cannot find the private key? I have triple checked I have it correct and it was working before swimmer?

File "/Users/simosa/Downloads/crabada.py/src/helpers/config.py", line 208, in validateUsersConfigs
    raise MissingConfig("Could not find user private key in config")
src.common.exceptions.MissingConfig: Could not find user private key in config

**Reply by OneArb** [reply]
Can you check in case you'd have a dangling

`USER_1_PRIVATE_KEY="" # as exported from your wallet - it's a hex string without the initial 0x, not the mnemonic seed phrase`

line in `.env`?

That second line will superseed a previous `USER_1_PRIVATE_KEY` statement.

**Reply by coccoinomane** [reply]
Hi @Simosa1080 !

If you still have issues after following @OneArb 's advice, you can try removing all single (`'`) and double (`''`) quotes from your _.env_.

Cheers,
Cocco

**Reply by Simosa1080** [reply]
got it thanks guys!
